pyproject.toml

```diff
@@ -0,0 +1,109 @@
+[build-system]
+requires = ['hatchling', 'hatch-vcs']
+build-backend = 'hatchling.build'
+
+[project]
+name = 'fmf'
+authors = [
+    { name = 'Petr Splichal', email = 'psplicha@redhat.com' },
+]
+maintainers = [
+    { name = 'Petr Splichal', email = 'psplicha@redhat.com' },
+]
+description = 'Flexible Metadata Format'
+readme = 'README.rst'
+license = 'GPL-2.0-or-later'
+license-files = { paths = ['LICENSE'] }
+requires-python = '>=3.9'
+classifiers = [
+    'Natural Language :: English',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
+    'Topic :: Utilities',
+]
+keywords = [
+    'metadata',
+    'testing',
+]
+dependencies = [
+    'ruamel.yaml',
+    'filelock',
+    'jsonschema',
+]
+dynamic = ['version']
+
+[project.urls]
+Homepage = 'https://github.com/teemtee/fmf'
+Documentation = 'https://fmf.readthedocs.io'
+
+[project.optional-dependencies]
+tests = [
```

Comment:
While it's been(or still is?) common practice to have tests, docs as optional extras, I'm personally in favor of not creating them, especially since there are hatch envs and the setup can be same/similar to the one in tmt.

Reply:
Sure, let me check that one again. One thing that might be missing is the coverage part. The only other usages are in Github Actions and RTD. The former I've completely separated for future design discussions, but what about the latter? It is simpler to configure it with:
```yaml
python:
    install:
      - method: pip
        path: .
        extra_requirements: [docs]
```
rather than a hatch approach, don't you think?

Reply:
So I'm not worried about python packaging part, but rather that it should be 1:1 to Fedora packaging, so we would be creating `fmf+docs`, `fmf+tests` rpms. Same way as in tmt there are ghost packages for plugins, e.g. `tmt+provision-container`.  

But in general, I think it would be good to keep fmf setup as close to tmt as possible, for reasons like having the same developer workflows, fixing potential issues simultaneously with same fixes, etc.

Reply:
`docs` is an outlier, yes. If the rtd workflow is the same here as in tmt, I assume we should do the same "workaround".  

> they have to be included manually

Right, but not including them is against the quidelines

> The question is, can we pip install .[docs] if it's inside a hatch environment?

https://github.com/teemtee/tmt/blob/b782a6067f4c2f8f2a10318edd3a6b06ccced5de/pyproject.toml#L173

Reply:
Oh, I was trying to migrate `test` when I hit another issue, in `fmf.spec` the pytest tests are still run, unlike `tmt` which runs all the tests in `testing-farm`. What do you suppose we do in this case?

Reply:
hmm, @psss @happz @lukaszachy, I don't know if I have the full context, so please correct me if I'm missing something, but I would say we should strive to have the same setup as in tmt, incl. how and where the tests are being run?

Reply:
For reference:

https://github.com/teemtee/fmf/blob/e31f7e7207c417cf84a10ff768da8e63a0ec7944/fmf.spec#L61-L62

Vs:

https://github.com/teemtee/tmt/blob/24d86c9ca56f952e1247b21bfe517fc331cc1cd4/tmt.spec#L130-L131

Reply:
(check import being default sanity check in Fedora)

Reply:
I'm thinking whether there is any benefit in running the tests again during rpm build. WDYT?  
imho let's just do the same as tmt.

Reply:
No strong opinions really. But in that case we should add a `gating.yaml` on the `fmf` (and maybe `tmt` as well) downstream.